packages/x-data-grid/src/utils/createSelector.ts

```diff
@@ -149,22 +144,24 @@ export const createSelectorMemoized: CreateSelectorFunction = (...args: any) =>
       }
     }
 
-    const { cache } = cacheContainer;
+    const cacheArgsInit = cache.get(cacheKey);
+    const cacheArgs = cacheArgsInit ?? new Map();
+    const cacheFn = cacheArgs?.get(args);
 
-    if (cache.get(cacheKey) && cache.get(cacheKey)?.get(args)) {
+    if (cacheArgs && cacheFn) {
       // We pass the cache key because the called selector might have as
       // dependency another selector created with this `createSelector`.
-      return cache.get(cacheKey)?.get(args)(state, cacheKey);
+      return cacheFn(state, cacheKey);
     }
```

Comment:
We're doing `map.get` 4 times but there are only 2 levels of depth. This is called very often.